multilabel_classification/pytorch_network_training.ipynb

```diff
@@ -0,0 +1,1064 @@
+    "\n",
+    "        labels = self.labels[idx]\n",
+    "\n",
+    "        return img, labels\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from collections import Counter\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = DatasetMultiLabel(df = df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = create_model(\n",
+    "    'efficientnet_b0',\n",
+    "    num_classes=len(dataset.labels[0]),\n",
+    ")\n",
+    "data_config = resolve_data_config(\n",
+    "       args = {}, model=model\n",
+    "    )\n",
+    "\n",
+    "model = MultiLabelModel(\n",
+    "        model,\n",
+    "        n_classes=len(dataset.labels[0]),\n",
+    "    ).cuda()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.model_selection import StratifiedKFold"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_splits = 4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skf = StratifiedKFold(n_splits=num_splits)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def reset_weights(m):\n",
+    "  '''\n",
+    "    Try resetting model weights to avoid\n",
+    "    weight leakage.\n",
+    "  '''\n",
+    "  for layer in m.children():\n",
+    "    if hasattr(layer, 'reset_parameters'):\n",
+    "#         print(f'Reset trainable parameters of layer = {layer}')\n",
+    "        layer.reset_parameters()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_df(pred_probs,dataset):\n",
+    "    ls = dataset_val.label_names\n",
+    "    cl = defaultdict(list)\n",
+    "    cl['image_loc'] = dataset.data\n",
+    "    for i in range(0,len(ls)):\n",
+    "        cl[ls[i]] = pred_val.T[i].tolist()\n",
+    "    return pd.DataFrame.from_dict(cl)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pred_val = model.predict_proba(loader_val)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from timm.utils import CheckpointSaver\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
```

Comment:
Most people will view this notebook through github's viewer:
https://github.com/cleanlab/examples/blob/fbb88a4055cb7f942520dc59409551a559de71dc/multilabel_classification/pytorch_network_training.ipynb

You can see this output cell is huge and takes forever to scroll through, so it would be good to print less output by default (you can add `verbosity` arg to your model for example and reduce the verbosity level to suppress constant printing of the loss)

Reply:
Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e